e2e/smoke_test.go

```diff
@@ -85,102 +86,121 @@ var _ = Describe("Smoke End To End Tests - against OpenShift Cluster with Istio
+
+				// and modify the service
+				modifiedDetails := strings.Replace(DetailsRuby, "PublisherA", "Publisher Ike", 1)
+				CreateFile(tmpDir+"/ratings.rb", modifiedDetails)
+
+				// then
+				Eventually(func() (string, error) {
+					fmt.Printf("[%s] checking...\n", time.Now().Format("2006-01-02 15:04:05.001"))
+					return GetBodyWithHeaders("http://istio-ingressgateway-istio-system.127.0.0.1.nip.io/productpage", map[string]string{"x-test-suite": "smoke"})
+				}, 3*time.Minute, 1*time.Second).Should(ContainSubstring("Publisher Ike"))
+
+				stopFailed := ikeWithWatch.Stop()
+				Expect(stopFailed).ToNot(HaveOccurred())
+
+				Eventually(ikeWithWatch.Done(), 1*time.Minute).Should(BeClosed())
+			})
+
+			It("should watch for changes in ratings service in specified namespace and serve it", func() {
+				doBasicScenarioCheck(tmpDir, namespace)
+			})
 		})
 
-		It("should watch for changes in ratings service in specified namespace and serve it", func() {
-			Eventually(AllPodsNotInState(namespace, "Running"), 3*time.Minute, 2*time.Second).
-				Should(ContainSubstring("No resources found"))
-
-			Eventually(func() (string, error) {
-				return GetBody("http://istio-ingressgateway-istio-system.127.0.0.1.nip.io/productpage")
-			}, 3*time.Minute, 1*time.Second).Should(ContainSubstring("ratings-v1"))
-
-			// switch to different namespace
-			<-cmd.Execute("oc project myproject").Done()
+		Context("scenario-2-basic-deploymentconfig", func() {
+			BeforeEach(func() {
+				scenario = "scenario-2"
+			})
 
-			// given we have details code locally
-			CreateFile(tmpDir+"/ratings.rb", DetailsRuby)
-
-			// when we start ike with watch
-			ikeWithWatch := cmd.ExecuteInDir(tmpDir, "ike", "develop",
-				"--deployment", "ratings-v1",
-				"-n", namespace,
-				"--port", "9080",
-				"--method", "inject-tcp",
-				"--watch",
-				"--run", "ruby ratings.rb 9080",
-				"--route", "header:x-test-suite=smoke",
-			)
-
-			// ensure the new service is running
-			Eventually(AllPodsNotInState(namespace, "Running"), 3*time.Minute, 2*time.Second).
-				Should(ContainSubstring("No resources found"))
-
-			// and modify the service
-			modifiedDetails := strings.Replace(DetailsRuby, "PublisherA", "Publisher Ike", 1)
-			CreateFile(tmpDir+"/ratings.rb", modifiedDetails)
-
-			// then
-			Eventually(func() (string, error) {
-				fmt.Printf("[%s] checking...\n", time.Now().Format("2006-01-02 15:04:05.001"))
-				return GetBodyWithHeaders("http://istio-ingressgateway-istio-system.127.0.0.1.nip.io/productpage", map[string]string{"x-test-suite": "smoke"})
-			}, 3*time.Minute, 1*time.Second).Should(ContainSubstring("Publisher Ike"))
-
-			stopFailed := ikeWithWatch.Stop()
-			Expect(stopFailed).ToNot(HaveOccurred())
-
-			Eventually(ikeWithWatch.Done(), 1*time.Minute).Should(BeClosed())
+			It("should watch for changes in ratings service in specified namespace and serve it", func() {
+				doBasicScenarioCheck(tmpDir, namespace)
```

Comment:
Maybe we can rename `doBasicScenarioCheck(tmpDir, namespace)` to `verifyThatResponseMatchesModifiedService(tmpDir, namespace)`?

Reply:
https://github.com/Maistra/istio-workspace/pull/95/commits/4ab788abfa01b34506adbdb49691823a43253f1c